File: lib/models/amap_business_area.dart
import 'package:amap_flutter_search/models/amap_geo_point.dart';
import 'package:json_annotation/json_annotation.dart';
part 'amap_business_area.g.dart';

@JsonSerializable()
class AMapBusinessArea {
  ///名称
  String? name;

  ///中心坐标
  AMapGeoPoint? location;

  AMapBusinessArea();
  factory AMapBusinessArea.fromJson(Map<String, dynamic> srcJson) =>
      _$AMapBusinessAreaFromJson(srcJson);

  Map<String, dynamic> toJson() => _$AMapBusinessAreaToJson(this);
}

File: android/src/main/java/com/amap/amap_flutter_search/AMapBusinessArea.java
package com.amap.amap_flutter_search;

public class AMapBusinessArea {
    ///名称
    private String name;

    ///中心坐标
    private AMapGeoPoint location;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public AMapGeoPoint getLocation() {
        return location;
    }

    public void setLocation(AMapGeoPoint location) {
        this.location = location;
    }
}
